src/app/producer-shelf/components/producer-shelf/producer-shelf.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, ElementRef, HostListener, OnDestroy, ViewChild } from '@angular/core'
+import { KeyboardBindingService } from '../../abstractions/keyboard-binding.service'
+import { KeyBinding } from '../../models/key-binding'
+import { KeyboardBindingMatcher } from '../../services/keyboard-binding.matcher'
+import { IconButton, IconButtonSize } from '../../../shared/enums/icon-button'
+
+@Component({
+  selector: 'sofie-producer-shelf',
+  templateUrl: './producer-shelf.component.html',
+  styleUrls: ['./producer-shelf.component.scss'],
+})
+export class ProducerShelfComponent implements OnDestroy {
```

Comment:
It looks like this component is responsible for both handling the dragging of the shelf and the content within the shelf.
It would be nice if these two things would be separated.

This can be achieved with `content injection` https://angular.io/guide/content-projection

We could then have one component incharge of the drag functionality, `DragableShelfComponent`, that we can inject content into.
In this case we would inject a `RundownViewShelfComponent` (or whatever name make sense) that knows how to display the keyboard etc.

Reply:
This was used to create the `DraggableShelf` component. The `ProducerShelf` will now be wrapped by the drag functionality.